judge/views/problem.py

```diff
@@ -581,9 +586,9 @@ def problem_submit(request, problem=None, submission=None):
                 raise Http404()
         if submission is not None:
             try:
-                sub = get_object_or_404(Submission, id=int(submission))
-                initial['source'] = sub.source
-                initial['language'] = sub.language
+                source, language = get_object_or_404(Submission.objects.values_list('source__source', 'language'), id=int(submission))
```

Comment:
Do I really have to point out that your line is too long every single time??? I thought you'd learn by now that the **line limit is 120 characters**.